### PR TITLE
Bring back test suite runs on GitHub actions

### LINKED ISSUE #1604: Reinstate test suite on GitHub Actions
A long time ago, we used to run our test suite on Travis. We've since moved to GitHub Actions, but are yet to migrate the tests over there. We need to create a GitHub Actions workflow that runs before the container builds that goes through the test suite.

Also, the below script below doesn't use containers, so we should modify the script to run in a containerized environment to mimic the real runtime environment.

## Old test script for reference
```yaml
language: python
python:
  - "2.7"
virtualenv:
  system_site_packages: true
addons:
  firefox: "45.0"
  apt:
    update: true
    packages:
      - net-tools
      - mplayer
      - ffmpeg
      - python-gobject
      - python-setuptools
install:
  - pip install -U pip
  - pip install -r requirements/requirements.txt
  - pip install -r requirements/requirements.dev.txt
before_script:
  - mkdir -p ~/.screenly ~/.config/uzbl/ ~/screenly_assets /tmp/USB/cleanup_folder
  - cp ansible/roles/screenly/files/uzbl-config ~/.config/uzbl/config-screenly
  - cp ansible/roles/screenly/files/screenly.conf ~/.screenly/
  - cp ansible/roles/screenly/files/screenly.db ~/.screenly/
  - sudo cp bin/install.sh /usr/local/sbin/upgrade_screenly.sh
  - sudo chmod 0700 /usr/local/sbin/upgrade_screenly.sh
  - echo -e "[local]\nlocalhost ansible_connection=local" > ansible/localhost
  - curl https://www.screenly.io/upload/ose-logo.png > /tmp/image.png
  - cp /tmp/image.png /tmp/USB/image.png
  - cp /tmp/image.png /tmp/USB/cleanup_folder/image.png
  - curl https://www.screenly.io/upload/ose-logo.png > ~/screenly_assets/image.tmp
  - curl https://www.screenly.io/upload/big_buck_bunny_720p_10mb.flv > /tmp/video.flv
  - cp tests/assets/asset.mov /tmp/asset.mov
  - export DISPLAY=:99.0
  - /sbin/start-stop-daemon --start --quiet --pidfile /tmp/custom_xvfb_99.pid --make-pidfile --background --exec /usr/bin/Xvfb -- :99 -ac -screen 0 1280x1024x16
  - sudo cp tests/config/ffserver.conf /etc/ffserver.conf
  - /usr/bin/ffserver -f /etc/ffserver.conf &
  - sleep 3
  - python server.py &
  - sleep 3 # give xvfb some time to start

script:
  - find . ! -path "*/rtmplite/*" -name \*.py -exec pep8 --ignore=E402,E501,E731 {} +
  - nosetests -v -a '!fixme'
  - ansible-playbook --syntax-check -i ansible/localhost ansible/site.yml
  - python -m SimpleHTTPServer 8081 &
  - sleep 3
  - bash static/spec/runner.sh

sudo: false
dist: xenial
```

### PR DESCRIPTION
**Pull request by nicomiguelino**
### General

- Might fix #1604.

### Running Tests Locally (Dockerized)

```bash
docker-compose -f docker-compose.test..yml up -d
docker-compose -f docker-compose.test.yml exec srly-ose-test bash bin/run_tests.sh
```

### Checklist

- [x] Modify the `.github/workflows/docker-build.yaml` file so that tests could be run before Docker builds.
- [x] Modify `README.md` and/or any relevant docs.

### CODE REVIEW

File: .github/workflows/docker-build.yaml [hunk 1]
```diff
@@ -131,5 +157,5 @@ jobs:
         id: build
         with:
           balena_token: ${{ secrets.BALENA_TOKEN }}
-          fleet: screenly_ose/screenly-ose-${{ matrix.board }}
```
Comment (vpetersson) [inline .github/workflows/docker-build.yaml:134]: Please revert this change. That was a change I made today.

File: docker/Dockerfile.test [hunk 2]
```diff
@@ -0,0 +1,20 @@
+FROM screenly/srly-ose-base:latest-x86
+
+ADD requirements/requirements.txt /tmp/requirements.txt
+RUN pip install --no-cache-dir -r /tmp/requirements.txt
+
+ADD requirements/requirements.dev.txt /tmp/requirements.dev.txt
+RUN pip install --no-cache-dir -r /tmp/requirements.dev.txt
+
+RUN pip install --no-cache-dir 'python-vlc==3.0.11115'
```
Comment (vpetersson) [inline docker/Dockerfile.test:9]: Isn't it better to pull in all requirements from viewer here instead?

**Review by vpetersson — APPROVED** [review]
Approved with minor comment. These tests ought to be fast since we're not running any of them :)

File: docker/Dockerfile.test (hunk 2 above)
Comment (nicomiguelino) [reply docker/Dockerfile.test:9]: I'll try using the viewer requirements instead. Thanks!

**Review by vpetersson — APPROVED** [review]
